identity-did/src/document/core_document.rs

```diff
@@ -256,6 +267,59 @@ impl<T, U, V> CoreDocument<T, U, V> {
     self.verification_method.remove(did);
   }
 
+  /// Attaches the relationship to the given method, if the method exists.
+  ///
+  /// Note: The method needs to be in the set of verification methods,
+  /// so it cannot be an embedded one.
+  pub fn attach_method_relationship<'query, Q>(
+    &mut self,
+    method_query: Q,
+    relationship: MethodRelationship,
+  ) -> Result<bool>
+  where
+    Q: Into<MethodQuery<'query>>,
+  {
+    let method: &VerificationMethod<_> = self
+      .resolve_method_with_scope(method_query, MethodScope::VerificationMethod)
+      .ok_or(Error::QueryMethodNotFound)?;
+
+    let method_ref = MethodRef::Refer(method.id().clone());
+
+    let was_attached = match relationship {
+      MethodRelationship::Authentication => self.authentication_mut().append(method_ref),
+      MethodRelationship::AssertionMethod => self.assertion_method_mut().append(method_ref),
+      MethodRelationship::KeyAgreement => self.key_agreement_mut().append(method_ref),
+      MethodRelationship::CapabilityDelegation => self.capability_delegation_mut().append(method_ref),
+      MethodRelationship::CapabilityInvocation => self.capability_invocation_mut().append(method_ref),
+    };
+
+    Ok(was_attached)
+  }
+
+  // Detaches the given relationship from the given method, if the method exists.
+  pub fn detach_method_relationship<'query, Q>(
```

Comment:
Nit: comment -> doc comment

```suggestion
  /// Detaches the given relationship from the given method, if the method exists.
  pub fn detach_method_relationship<'query, Q>(
```